### ISSUE TITLE
auto-sklearn classifier returning training score of 1.0

### ISSUE THREAD

**Issue by tim37123** [automl/auto-sklearn#162]
I'm running a set of data through auto-sklearn classifier, watching it run I seem to get some promising results, I've seen it indicate the training data move north of .9 which is better than I've been able to achieve in my own model tweaking, however it seems to continue to run for hours after that and eventually starts returning 'Training performance: 1.000000'. When I try to score a test set against the model I get '-1'. This has happened all 3 times I've tried to fit. Any idea why this can happen? I've used the train/test set against sklearn classifiers directly without a problem. I've attached the output of the run [here](https://drive.google.com/file/d/0B2Z-d9ssGRjzSnJJV0FfajcwSUU/view?usp=sharing)

**Comment by mfeurer** [comment]
Hi, Thanks for your interest in auto-sklearn. It seem like you are using four datasets with 1023, 78, 45 and 6 data points. Did I parse the logs correctly? Out of those, which has the test score of '-1'? In any case, since your datasets are very small, I advise you to use 10-fold cross-validation to avoid overfitting, which might be the problem here. You can find an example in the [auto-sklearn repository](https://github.com/automl/auto-sklearn/blob/master/example/example_crossvalidation.py). Let me know if this helps.

**Comment by tim37123** [comment]
Thank you. I'm using train_test_split to break the dataset into dataset into train and test groups, but those splits sound feasible. I've restructured to account for cross validation and seem to be getting better results.

**Comment by mfeurer** [comment]
Great to hear. Keep in mind that we never tested auto-sklearn with a dataset with less than 667 training data points.

I'm closing this now; if you have any more questions, please reopen this issue or create a new one.
